app/controllers/developers_controller.rb

```diff
@@ -41,7 +41,13 @@ def update
   end
 
   def show
-    @developer = find_developer!
+    finder = Developers::Finder.new(id: params[:id]).call
+    @developer = finder.developer
+
+    if finder.should_redirect?
+      redirect_to @developer, status: 302, notice: t(".redirection", url: developer_url(@developer))
+    end
+
```

Comment:
Nice! I like how you pulled out the redirection question.

But I don't think we need the `#call` method. We could have `#developer` and `#should_redirect?` memoize so no matter which one you call first you get the same behavior.

For example,

```ruby
class Finder
  # ...

  def developer
    find_developer unless defined?(@developer)
    @developer
  end

  def should_redirect?
    find_developer unless defined?(@redirect)
    @redirect
  end

  private

  def find_developer
    # logic of #call
    # set @developer and @redirect
  end
end
```

Reply:
Interesting, never thought to memoize this way. Cool stuff!